core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -213,17 +214,19 @@ public Response getBucket(@HeaderParam("Authorization") String authorization,
                             @QueryParam("prefix") final String prefixParam,
                             @QueryParam("delimiter") final String delimiterParam,
                             @QueryParam("encoding-type") final String encodingTypeParam,
-                            @QueryParam("max-keys") final int maxKeysParam,
+                            @QueryParam("max-keys") final String maxKeysParam,
                             @QueryParam("list-type") final int listTypeParam,
                             @QueryParam("continuation-token") final String continuationTokenParam,
                             @QueryParam("start-after") final String startAfterParam,
                             @QueryParam("tagging") final String tagging) {
     return S3RestUtils.call(bucket, () -> {
       Preconditions.checkNotNull(bucket, "required 'bucket' parameter is missing");
+      // TODO(czhu): move parameter validation into constructor for ListBucketOptions
       String marker = markerParam == null ? "" : markerParam;
       String prefix = prefixParam == null ? "" : prefixParam;
       String encodingType = encodingTypeParam == null ? "url" : encodingTypeParam;
-      int maxKeys = maxKeysParam <= 0 ? ListBucketOptions.DEFAULT_MAX_KEYS : maxKeysParam;
+      int maxKeys = maxKeysParam == null ? ListBucketOptions.DEFAULT_MAX_KEYS
+          : Math.max(Integer.parseInt(maxKeysParam), 0);
```

Comment:
Can you transform the number format exception to an S3 exception here? I think it is an InvalidArgument exception code https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html

Reply:
That's correct, I'm not sure why the prior implementation was silently converting the result to the default value but that's why I followed suit and added the comment. I'll make the change here then.

Reply:
I mean catch the NumberFormatException and change it to an S3 exception with the correct code. Is that happening somewhere?

Reply:
Resolving as outdated